caddyhttp/staticfiles/fileserver.go

```diff
@@ -40,6 +40,14 @@ func (fs FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, err
 	return fs.serveFile(w, r, r.URL.Path)
 }
 
+// calculateEtag produces a strong etag by default. Prefix the result with "W/" to convert this into a weak one.
+// see https://tools.ietf.org/html/rfc7232#section-2.3
+func calculateEtag(d os.FileInfo) string {
+	t := strconv.FormatInt(d.ModTime().Unix(), 36)
+	s := strconv.FormatInt(d.Size(), 36)
+	return fmt.Sprintf(`"%s%s"`, t, s)
+}
```

Comment:
This is my only nit-pick, and it is a nit-pick. :) I suspect `t + s` will be slightly faster than using the reflection features of `fmt.Sprintf`. Might be worth changing it to `return t + s`.